pkg/apis/core/validation/exposureclass.go

```diff
@@ -17,17 +17,40 @@ package validation
 import (
 	"github.com/gardener/gardener/pkg/apis/core"
 
+	apivalidation "k8s.io/apimachinery/pkg/api/validation"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 )
 
 // ValidateExposureClass validates a ExposureClass object.
 func ValidateExposureClass(exposureClass *core.ExposureClass) field.ErrorList {
-	allErrs := field.ErrorList{}
+	var allErrs = field.ErrorList{}
+
+	if len(exposureClass.Handler) == 0 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("handler"), exposureClass.Name, "exposure class handler cannot be empty"))
+	}
+
+	return append(allErrs, validateExposureClassScheduling(exposureClass.Scheduling, field.NewPath("scheduling"))...)
+}
+
+func validateExposureClassScheduling(scheduling *core.ExposureClassScheduling, fldPath *field.Path) field.ErrorList {
+	var allErrs = field.ErrorList{}
+
+	if scheduling == nil {
+		return append(allErrs, field.Required(fldPath, "exposure class need to have a scheduling configuration"))
+	}
+
+	if (scheduling.SeedSelector == nil && len(scheduling.Tolerations) == 0) || (scheduling.SeedSelector != nil && len(scheduling.Tolerations) > 0) {
```

Comment:
What about properly validating the tolerations like for `Shoot`s (https://github.com/gardener/gardener/blob/f5de6826ec488d56f37566a71613b0ae90a414b7/pkg/apis/core/validation/shoot.go#L156)?

Reply:
done